### PR TITLE
Fix tripleo network settings update steps

### PR DESCRIPTION
**Pull request by booxter**
These are reconstructed as per (sorry for the long url...):

https://docs.redhat.com/en/documentation/red_hat_openstack_platform/17.1 /html/installing_and_managing_red_hat_openstack_platform_with_director /assembly_provisioning-and-deploying-your-overcloud# proc_provisioning-bare-metal-nodes-for-the-overcloud_ironic_provisioning

### CODE REVIEW

**Comment by booxter** [comment]
@hjensas could you please check this makes sense, as per your comments [elsewhere](https://github.com/openstack-k8s-operators/data-plane-adoption/pull/641#discussion_r1805124736)? Thanks.

File: docs_user/modules/proc_using-new-subnet-ranges.adoc [hunk 1]
```diff
@@ -35,9 +35,19 @@ Repeat this configuration for other networks that need to use different subnets
...
+ /home/stack/templates/<networks_definition_file>
+----
++
+----
+(undercloud)$ openstack overcloud node provision \
+ [--templates <templates_directory> \]
+ --stack <stack> \
+ --network-config \
+ --output <deployment_file> \
+ /home/stack/templates/<node_definition_file>
```
Comment (klgill) [inline docs_user/modules/proc_using-new-subnet-ranges.adoc:50]: We also need the list that describes the values to be replaced from step 10 in [1].
I can do this in another PR if you want. But please confirm that the items below the command in step 10[1] are relevant for this step in "Configuring new subnet ranges".

[1] https://docs.redhat.com/en/documentation/red_hat_openstack_platform/17.1/html/installing_and_managing_red_hat_openstack_platform_with_director/assembly_provisioning-and-deploying-your-overcloud#proc_provisioning-bare-metal-nodes-for-the-overcloud_ironic_provisioning

Comment (hjensas) [reply docs_user/modules/proc_using-new-subnet-ranges.adoc:50]: When running `overcloud node provision` - we also need to toggle the `network_config_update` for each role in the `<node_definition_file>`. Ref the [4.11. Adding a new leaf to a spine-leaf deployment](https://docs.redhat.com/en/documentation/red_hat_openstack_platform/17.1/html/configuring_spine-leaf_networking/assembly_configuring-the-overcloud#proc_add-new-leaf_spine-leaf)[1] doc, step 9.

I think you can add that in the follow up as well.

https://docs.redhat.com/en/documentation/red_hat_openstack_platform/17.1/html/configuring_spine-leaf_networking/assembly_configuring-the-overcloud#proc_add-new-leaf_spine-leaf

**Review by hjensas — COMMENTED** [review]
LGTM, minor comment - but can be done as part of the already mentioned follow up.

**Review by jistr — APPROVED** [review]

**Comment by klgill** [comment]
I'm merging this PR and adding the feedback in a separate PR.

**Pull request merged**